src/cryptenroll/cryptenroll-pkcs11.c

```diff
@@ -35,9 +34,9 @@ int enroll_pkcs11(
         if (r < 0)
                 return r;
 
-        r = X509_certificate_generate_volume_key(cert, &decrypted_key, &decrypted_key_size, &savedata, &savedata_size);
+        r = X509_certificate_generate_volume_key(cert, &decrypted_key, &decrypted_key_size, &saved_key, &saved_key_size);
         if (r < 0)
-                return log_error_errno(r, "Failed to generate secret key for X.509 certifcate: %m");
+                return r;
```

Comment:
hmm, this does what i asked for in the earlier patch. please adjust the earlier patch to not make this change in the first place.

we generally want "perfect" PRs, where each patch makes logical sense on its own, is perfectly bisectable and doesn't "fix" changes made in earlier commits of the PR. We do not care about historical sense so much.